client/pluginmanager/drivermanager/manager_test.go

```diff
@@ -0,0 +1,300 @@
+package drivermanager
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	log "github.com/hashicorp/go-hclog"
+	plugin "github.com/hashicorp/go-plugin"
+	"github.com/hashicorp/nomad/client/pluginmanager"
+	"github.com/hashicorp/nomad/client/state"
+	"github.com/hashicorp/nomad/helper/testlog"
+	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/hashicorp/nomad/plugins/base"
+	"github.com/hashicorp/nomad/plugins/drivers"
+	"github.com/hashicorp/nomad/plugins/shared/loader"
+	"github.com/hashicorp/nomad/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+var _ Manager = (*manager)(nil)
+var _ pluginmanager.PluginManager = (*manager)(nil)
+
+func testSetup(t *testing.T) (chan *drivers.Fingerprint, chan *drivers.TaskEvent, *manager) {
+	fpChan := make(chan *drivers.Fingerprint)
+	evChan := make(chan *drivers.TaskEvent)
+	drv := mockDriver(fpChan, evChan)
+	cat := mockCatalog(map[string]drivers.DriverPlugin{"mock": drv})
+	cfg := &Config{
+		Logger:       testlog.HCLogger(t),
+		Loader:       cat,
+		PluginConfig: nil,
+		Updater:      noopUpdater,
+		State:        state.NoopDB{},
+	}
+
+	mgr := New(cfg)
+	return fpChan, evChan, mgr
+}
+
+func mockDriver(fpChan chan *drivers.Fingerprint, evChan chan *drivers.TaskEvent) drivers.DriverPlugin {
+	return &drivers.MockDriver{
+		FingerprintF: func(ctx context.Context) (<-chan *drivers.Fingerprint, error) {
+			return fpChan, nil
+		},
+		TaskEventsF: func(ctx context.Context) (<-chan *drivers.TaskEvent, error) {
+			return evChan, nil
+		},
+	}
+}
+
+func mockCatalog(drivers map[string]drivers.DriverPlugin) *loader.MockCatalog {
+	cat := map[string][]*base.PluginInfoResponse{
+		base.PluginTypeDriver: {},
+	}
+	for d := range drivers {
+		cat[base.PluginTypeDriver] = append(cat[base.PluginTypeDriver], &base.PluginInfoResponse{
+			Name: d,
+			Type: base.PluginTypeDriver,
+		})
+	}
+
+	return &loader.MockCatalog{
+		DispenseF: func(name, pluginType string, cfg *base.ClientAgentConfig, logger log.Logger) (loader.PluginInstance, error) {
+			d, ok := drivers[name]
+			if !ok {
+				return nil, fmt.Errorf("driver not found")
+			}
+			return loader.MockBasicExternalPlugin(d), nil
+		},
+		ReattachF: func(name, pluginType string, config *plugin.ReattachConfig) (loader.PluginInstance, error) {
+			d, ok := drivers[name]
+			if !ok {
+				return nil, fmt.Errorf("driver not found")
+			}
+			return loader.MockBasicExternalPlugin(d), nil
+		},
+		CatalogF: func() map[string][]*base.PluginInfoResponse {
+			return cat
+		},
+	}
+}
+
+func mockTaskEvent(taskID string) *drivers.TaskEvent {
+	return &drivers.TaskEvent{
+		TaskID:      taskID,
+		Timestamp:   time.Now(),
+		Annotations: map[string]string{},
+		Message:     "event from " + taskID,
+	}
+}
+
+func noopUpdater(string, *structs.DriverInfo) *structs.Node { return nil }
+
+func TestMananger_FingerPrint(t *testing.T) {
```

Comment:
```suggestion
func TestMananger_Fingerprint(t *testing.T) {
```

I know this is the ultimate nitpick, but I swear I've had trouble finding a Fingerprint related thing before because I was doing a case sensitive search and the P was uppercased. :sweat_smile: